api/src/neo4j/queries/interactionPartners.js

```diff
@@ -21,41 +21,32 @@ YIELD value as v
 
 WITH v.component as component
 MATCH (${m} {id: component.id})-[${v}]-(r:Reaction)
+WITH component, collect(r) as reactions
+
+CALL apoc.cypher.mapParallel2("
+ WITH (_) as reaction
+ MATCH (reaction)-[${v}]-(cm:CompartmentalizedMetabolite)
```

Comment:
This is suprising to me, since it seems there is absolutely no filter on the starting node. I'm note sure if it's meant to be like this. I guess what I intuitively was expecting to see was something of the sort `(reaction:Reaction)`, meaning that the starting `reaction` node must have the label `:Reaction`.

Reply:
```
WITH v.component as component
MATCH (${m} {id: component.id})-[${v}]-(r:Reaction)
WITH component, collect(r) as reactions
```

`CALL apoc.cypher.mapParallel2(querystring, {}, reactions, 50) YIELD value)`

The starting node is an item in the list called reactions, which is created by the code copy pasted above (line 22-24) . So they do have they labels

Reply:
Ah thanks 👍🏻